mozci/push.py

```diff
@@ -273,13 +273,15 @@ def tasks(self):
         except MissingDataError:
             pass
 
-        tasks = [Task.create(push=self, **task) for task in tasks]
+        tasks = [Task.create(**task) for task in tasks]
 
         # Gather group data.
         logger.debug(f"Gathering test groups for {self.rev}...")
         concurrent.futures.wait(
             [
-                Push.THREAD_POOL_EXECUTOR.submit(lambda task: task.groups, task)
+                Push.THREAD_POOL_EXECUTOR.submit(
+                    lambda task: task.retrieve_results(self), task
```

Comment:
I get why you're setting it up this way, and I guess we can land this as is for now to unblock things..

But I think I'd prefer storing `rev` and `branch` on the Task object since that will allow consumers to instantiate task objects directly. We could even create a new contract to grab `rev` / `branch` from taskcluster in the event that it wasn't filled in from the `create`. This would allow consumers to do stuff like:
```
task = Task(<id>)
print(task.results)
```

without requiring the context of a `Push` class to use. While we don't currently have consumers doing this, I'm often sent a taskcluster link and asked to investigate the given task id, so could imagine it being useful in that case. I could imagine other non-human consumers using this as well.

It can't increase the cache size *that* much can it? Are we currently pickling the `Push` class once per task and storing it in cache? If so storing just the `rev` / `branch` is already a big improvement :p

Reply:
We can definitely try and see how large it is before/after with a bunch of pushes.

With the `Push` reference, it's likely that we were storing the Push object only once for all tasks belonging to a push. IDK if Pickle will do the same for primitive values such as strings.